src/module_5/module5.md

```diff
@@ -0,0 +1,5880 @@
+    2006-12-30T00:00:00.000000000
+    2007-03-31T00:00:00.000000000
+    2007-06-30T00:00:00.000000000
+    2007-09-30T00:00:00.000000000
+    2007-12-30T00:00:00.000000000
+    2008-03-31T00:00:00.000000000
+    2008-06-30T00:00:00.000000000
+    2008-09-30T00:00:00.000000000
+    2008-12-30T00:00:00.000000000
+    2009-03-31T00:00:00.000000000
+    2009-06-30T00:00:00.000000000
+    2009-09-30T00:00:00.000000000
+    2009-12-30T00:00:00.000000000
+    2010-03-31T00:00:00.000000000
+    2010-06-30T00:00:00.000000000
+    2010-09-30T00:00:00.000000000
+    2010-12-30T00:00:00.000000000
+    2011-03-31T00:00:00.000000000
+    2011-06-30T00:00:00.000000000
+    2011-09-30T00:00:00.000000000
+    2011-12-30T00:00:00.000000000
+    2012-03-31T00:00:00.000000000
+    2012-06-30T00:00:00.000000000
+    2012-09-30T00:00:00.000000000
+    2012-12-30T00:00:00.000000000
+    2013-03-31T00:00:00.000000000
+    2013-06-30T00:00:00.000000000
+    2013-09-30T00:00:00.000000000
+    2013-12-30T00:00:00.000000000
+    2014-03-31T00:00:00.000000000
+    2014-06-30T00:00:00.000000000
+    2014-09-30T00:00:00.000000000
+    2014-12-30T00:00:00.000000000
+    2015-03-31T00:00:00.000000000
+    2015-06-30T00:00:00.000000000
+    2015-09-30T00:00:00.000000000
+    2015-12-30T00:00:00.000000000
+    2016-03-31T00:00:00.000000000
+    2016-06-30T00:00:00.000000000
+    2016-09-30T00:00:00.000000000
+    2016-12-30T00:00:00.000000000
+    2017-03-31T00:00:00.000000000
+    2017-06-30T00:00:00.000000000
+    2017-09-30T00:00:00.000000000
+    2017-12-30T00:00:00.000000000
+    2018-03-31T00:00:00.000000000
+    2018-06-30T00:00:00.000000000
+    2018-09-30T00:00:00.000000000
+    2018-12-30T00:00:00.000000000
+    2019-03-31T00:00:00.000000000
+    2019-06-30T00:00:00.000000000
+    2019-09-30T00:00:00.000000000
+    2019-12-30T00:00:00.000000000
+    2020-03-31T00:00:00.000000000
+    2020-06-30T00:00:00.000000000
+    2020-09-30T00:00:00.000000000
+    2020-12-30T00:00:00.000000000
+    2021-03-27T00:00:00.000000000
+
+
+
+```python
+learning_rates_train = compute_learning_rate("training", all_results)
+learning_rates_test = compute_learning_rate("valid_0", all_results)
+
+learning_rates_train = learning_rates_train[learning_rates_train["n_trees"] <= 20]
+learning_rates_test = learning_rates_test[learning_rates_test["n_trees"] <= 20]
+
+learning_rates_train = learning_rates_train[
+    ["n_trees", "normalized_learning_training", "execution_date"]
+]
+learning_rates_test = learning_rates_test[
+    ["n_trees", "normalized_learning_valid_0", "execution_date"]
+]
+
+learning_rates_train.rename(
+    columns={"normalized_learning_training": "Normalized Learning"}, inplace=True
+)
+learning_rates_test.rename(
+    columns={"normalized_learning_valid_0": "Normalized Learning"}, inplace=True
+)
+
+learning_rates_train["Set"] = "Training"
+learning_rates_test["Set"] = "Validation"
+
+learning_rates_combined = pd.concat([learning_rates_train, learning_rates_test])
```

Comment:
If you concatenate at the beginning you don't need to do the same filters / modifications for train and test